dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ExclusionFiles.java

```diff
@@ -0,0 +1,256 @@
+ *       child element.
+ *   <li>A TargetMatcher matches a linkage error when the target symbol (class, method, or field) of
+ *       the error matches its child element.
+ *   <li>A PackageMatcher matches the classes that have Java package specified by its name field.
+ *       Prefix to specify child packages.
+ *   <li>A ClassMatcher matches the class specified by its name attribute. ArtifactMatcher,
+ *       PackageMatcher, and ClassMatcher also match methods and fields on their matching classes.
+ *   <li>A MethodMatcher matches method symbol specified by className and name attribute.
+ *   <li>A FieldMatcher matches field symbol specified by className and name attribute.
+ * </ul>
+ */
+class ExclusionFiles {
+  private static final XMLEventFactory eventFactory = XMLEventFactory.newInstance();
+
+  private static final QName LINKAGE_CHECKER_FILTER_TAG = QName.valueOf("LinkageCheckerFilter");
+  private static final QName CLASS_TAG = QName.valueOf("Class");
+  private static final QName LINKAGE_ERROR_TAG = QName.valueOf("LinkageError");
+  private static final QName TARGET_TAG = QName.valueOf("Target");
+  private static final QName SOURCE_TAG = QName.valueOf("Source");
+  private static final QName METHOD_TAG = QName.valueOf("Method");
+  private static final QName FIELD_TAG = QName.valueOf("Field");
+
+  static ImmutableList<LinkageErrorMatcher> parse(Path exclusionFile)
+      throws SAXException, IOException, VerifierConfigurationException {
+
+    InputSource inputSource = new InputSource(Files.newInputStream(exclusionFile));
+    inputSource.setSystemId(exclusionFile.toUri().toString());
+
+    return parse(inputSource);
+  }
+
+  static ImmutableList<LinkageErrorMatcher> parse(URL exclusionFile)
+      throws SAXException, IOException, VerifierConfigurationException {
+
+    InputSource inputSource = new InputSource(exclusionFile.openStream());
+    inputSource.setSystemId(exclusionFile.toString());
+
+    return parse(inputSource);
+  }
+
+  private static ImmutableList<LinkageErrorMatcher> parse(InputSource inputSource)
+      throws SAXException, IOException, VerifierConfigurationException {
+
+    XMLReader reader = createXmlReader();
+
+    ExclusionFileHandler handler = new ExclusionFileHandler();
+    reader.setContentHandler(handler);
+
+    reader.parse(inputSource);
+
+    return handler.getMatchers();
+  }
+
+  private static XMLReader createXmlReader()
+      throws SAXException, IOException, VerifierConfigurationException {
+    // Validate and parse XML files in one pass using Jing validator as a filter.
+    // http://iso-relax.sourceforge.net/JARV/JARV.html#use_42
+    VerifierFactory factory = VerifierFactory.newInstance("http://relaxng.org/ns/structure/1.0");
+    InputStream linkageCheckerSchema =
+        ExclusionFiles.class
+            .getClassLoader()
+            .getResourceAsStream("linkage-checker-exclusion-relax-ng.xml");
```

Comment:
By convention this file name would end in .rng; that is, "linkage-checker-exclusion.rng"

https://en.wikipedia.org/wiki/RELAX_NG#Filename_extensions